utilities/bin/core/commands.dsc

```diff
@@ -0,0 +1,503 @@
+    aliases:
+        - dchant
+    allowed help:
+        - determine <proc[lib_has_permission].context[utilities.denchant|<player.if_null[null]>].if_null[false]>
+    tab completions:
+        1: <server.enchantments.parse[name]>
+        2: <proc[lib_numeric_list].context[<server.enchantment_start_level[<context.args.first>]>|<server.enchantment_max_level[<context.args.first>]>]>
+    script:
+        - if <proc[lib_has_permission].context[utilities.denchant|<player.if_null[server_or_command_block]>].if_null[true]>:
+            - if <player.exists>:
+                - if <context.args.size> <= 2:
+                    - if <context.args.size> >= 1:
+                        - define enchantment <enchantment[<context.args.first>].if_null[null]>
+                        - if <[enchantment]> != null:
+                            - define level <context.args.get[2].if_null[1]>
+                            - define colors <script[lib_config].parsed_key[color]>
+                            - define soft <[colors].get[soft_server_notice]>
+                            - define hard <[colors].get[hard_server_notice]>
+                            - define warn <[colors].get[warning]>
+                            - define item <player.item_in_hand>
+                            - define item_enchants <[item].enchantment_map.if_null[<map>]>
+                            - define prev_enchant <[item_enchants].get[<[enchantment].name>].if_null[null]>
+                            - if <[prev_enchant]> == null || <[prev_enchant]> != <[level]>:
+                                - define item_enchants <[item_enchants].exclude[<[enchantment].name>]> if:<[prev_enchant].equals[null].not>
+                                #books, which get turned into enchanted books, are always valid, regardless of what can_enchant says
+                                - define always_valid true if:<[item].advanced_matches[*book]>
+                                - if <[item].material.name> == book:
+                                    - inventory adjust slot:<player.held_item_slot> material:enchanted_book
+                                - inventory adjust slot:<player.held_item_slot> enchantments:<[item_enchants].with[<[enchantment]>].as[<[level]>]>
+                                - inventory adjust slot:<player.held_item_slot> lore:<[item].lore.if_null[<list>].include[<gray><[enchantment].full_name[<[level]>]>]> if:<[enchantment].key.before[:].equals[minecraft].not>
+                                - narrate "<[colors].get[success]>You successfully enchanted a <[soft]><[item].material.translated_name><[colors].get[success]> with <[colors].get[important]><[enchantment].full_name[<[level]>]>"
+                                - foreach <[item_enchants]>:
+                                    - define ench <enchantment[<[key]>]>
+                                    - define conflicts:->:<[ench].full_name> if:<[enchantment].is_compatible[<[ench]>].not>
```

Comment:
`enchantmenttag.full_name[<level>]`  requires level input, even if it's just the minimum level
```suggestion
                                    - define conflicts:->:<[ench].full_name[1]> if:<[enchantment].is_compatible[<[ench]>].not>
```

Reply:
Already fixed.